libraries/AP_HAL_Linux/ToneAlarm_Raspilot.cpp

```diff
@@ -181,16 +181,9 @@ void ToneAlarm_Raspilot::_set_pwm0_period(uint32_t time_us)
 void ToneAlarm_Raspilot::_set_pwm0_duty(uint8_t percent)
 {
     int bit_count = constrain_int32(320 * percent / 100, 320, 0);
-    unsigned int bits = 0;
 
     // FIXME: bits overflows for any bit_count > 32
-    while (bit_count) {
-      bits <<= 1;
-      bits |= 1;
-      bit_count--;
-    }
-
-    *(_pwm + RPI_PWM_DAT1) = bits;
+    *(_pwm + RPI_PWM_DAT1) = (1U << bit_count) - 1;
```

Comment:
@lucasdemarchi This patch isn't intended to fix the overflow (that is why it is still there btw), just get rid of the useless loop. Code is equivalent but better. Won't try to fix something I can test. Actually 32 is not even the right number, it may be 16 depending on what is installed/compilied againt. But if you just trunc at a magic number wont be the same that overflowing ? So what is the point anyway.